### DISCUSSION TITLE
Why ...args isn't transpiled to Array.prototype.slice.call(arguments)?

### DISCUSSION THREAD

**Discussion by wojtekmaj** [babel/babel#12667] [Q&A] [answered]
When I write:

```js
function foo(...args) {
  // …
}
```

it gets transpiled to 

```js
function foo() {
  for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
    args[_key] = arguments[_key];
  }

  // …
}
```

the same could be achieved by

```js
function foo() {
  var args = Array.prototype.slice.call(arguments);

  // …
}
```

so my question is - why? Is it some spec quirk? Is it safer? What's the benefit of using for loop here?

**Comment by JLHwung** [comment]
See https://stackoverflow.com/a/2143894/1490357 technically `arguments` is _not_ an array, it supports only indexing and a `.length` property. Although latest ES requires that `arguments` be iterable, this is not the case for IE.

**Reply by wojtekmaj** [reply]
Interesting! Thanks for quick response.

Just tested that on IE 11 and I believe your comment must apply to some older versions of IE, because this works like a charm:

![obraz](https://user-images.githubusercontent.com/5426427/105227722-ef51c180-5b61-11eb-98be-b934d5520f42.png)

Given that IE overall market share is at 1.05% as of now ([source](https://gs.statcounter.com/)), of which I suppose vast majority is version 11 on which the above works just fine, I think it would make sense to create browserslistrc-based solution which turns this "unsafe" transformation on and shaves off quite a few bytes for those of us who don't need to support anything older than IE 11.

What do you think?

**Comment by nicolo-ribaudo** [comment, marked as answer]
IIRC one reason is that passing `arguments` to another function causes a perf drop in some V8 versions.

**Reply by nicolo-ribaudo** [reply]
https://github.com/vhf/v8-bailout-reasons/#bad-value-context-for-arguments-value

It only affects Node < 8.3 and Chromium < 59, but for newer versions you don't need to compile rest parameters anyway.
